File: src/2015/day1/day1_part2.py
#!/usr/bin/env python3

with open("2015_day1.txt") as f:
  data = f.read().strip()

floor_n = 0
for i, char in enumerate(data):
  if (char == "("):
    floor_n += 1
  elif (char == ")"):
    floor_n -= 1
  if floor_n == -1:
    print(f"Reached basement at character position {i + 1}!!")
    break

File: src/2015/day1/day1_part2.js
const path = "./2015_day1.txt"
const file = Bun.file(path)
const text = await file.text();
let floor_n = 0;
for (let i = 0; i < text.length; i++) {
	if (text[i] == '(') {
		floor_n++;
	} else if (text[i] == ')') {
		floor_n--;
	}
	if (floor_n == -1) {
		console.log(`Step no: ${i}`);
		break;
	}
}